lib/spack/spack/cmd/solve.py

```diff
@@ -23,13 +24,13 @@
 level = 'long'
 
 #: output options
-show_options = ('asp', 'output', 'solutions')
+show_options = ('asp', 'opt', 'output', 'solutions')
 
 
 def setup_parser(subparser):
     # Solver arguments
     subparser.add_argument(
-        '--show', action='store', default=('solutions'),
+        '--show', action='store', default='opt,solutions',
```

Comment:
Minor nitpick. We need either to remove `opt` from the defaults or update the help:
```
  --show SHOW           outputs: a list with any of: asp, opt, output, solutions (default), all
```
to avoid a mismatch (since the default is to show optimization values).

Reply:
This is unrelated, but if you fix help strings can you also fix:
```
  -j, --json            print concrete spec as YAML
```
(should be JSON, not YAML)

Reply:
These are addressed.